sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/dao/room/EntityTypeConverters.kt

```diff
@@ -177,6 +201,77 @@ class EntityTypeConverters {
     }
 }
 
+/**
+ * A client writable copy is considered a new object with only the properties set on the object
+ * that are writable on bridge.  All other fields sent to bridge will be ignored anyways.
+ * @return a ScheduledActivity that can be sent to bridge
+ */
+fun ScheduledActivityEntity.clientWritableCopy(): ScheduledActivity {
+    val schedule = ScheduledActivity()
+    schedule.guid = guid
+    schedule.startedOn = startedOn?.let { org.joda.time.DateTime(it.toEpochMilli()) }
+    schedule.finishedOn = finishedOn?.let { org.joda.time.DateTime(it.toEpochMilli()) }
+    schedule.clientData = clientData?.data
+    return schedule
+}
+
+/**
+ * A bridge metadata copy is considered a new object with only the properties set on the object
+ * that the ArchiveUtil.createMetaDataFile() function is concerned with.
+ * @return a ScheduledActivity that can be pass to the BridgeDataProvider and have the metadata JSON included.
+ */
+fun ScheduledActivityEntity.bridgeMetadataCopy(): ScheduledActivity {
+    var schedule = ScheduledActivity()
+    val immutableFieldMap = HashMap<String, String>()
+
+    scheduledOn?.let {
+        // TODO: mdephillips 10/14/18 better way to convert from 3tenbp LocalDateTime to jodatime DateTime?
+        val dateTime = DateTime.now()
```

Comment:
val epochMilli = scheduledOn.toInstant().toEpochMilli()
val dateTime = DateTime(epochMilli, DateTimeZone.forID(scheduledOn.zone))

Reply:
Sweet, let me try this.  My way is pretty ridiculous.

Reply:
That didn't really compile.  toInstant takes a timezone as a parameter, plus there is no zone for scheduleOn because it is a LocalDateTime

Reply:
Oops, I thought scheduledOn was a ZonedDateTime. need to first to scheduledOn.atZone(ZoneId.systemDefault())
(We're defaulting in a timezone in the DateTime anyway)

Reply:
I'm fairly certain scheduledOn.atZone(ZoneId.systemDefault()) will move the time 5 hours (or whatever it is) no?

Reply:
I think scheduledOn would be 2018/10/16 08:00 (no zone, since it's a LocalDateTime), calling azZone(ZoneId.systemDefault()) turns it into a ZonedDateTime 2018/10/16 08:00-7:00, and the DateTime should also end up as 2018/10/16 08:00-7:00

Reply:
oh! perfect. that is what I want